### DISCUSSION TITLE
CMAKE Built task as root

### DISCUSSION THREAD

**Discussion by trolledbypro** [microsoft/vscode-cmake-tools#3285] [Ideas]
Greetings, I have several CMAKE projects (such as https://github.com/mavlink/MAVSDK) that require root access to build the project with the install target, as they install system wide into "C:\Program Files" and /usr/local/lib on Windows and Linux respectively.

Is there a possibility of having a feature where the CMAKE build task defined in tasks.json can have a "asRoot" option where the t ask can be run as root (and password entered as if you run a normal vscode task as sudo)?

**Comment by Zingam** [comment]
I think this is related to some other feature requests in cpptools and cmake-tools:
https://github.com/microsoft/vscode-cpptools/discussions/10863
https://github.com/microsoft/vscode-cmake-tools/issues/532
https://github.com/microsoft/vscode-cmake-tools/issues/352
https://github.com/microsoft/vscode-cpptools/issues/3155

You should probably create a feature request ticket and comment and upvote the related once. I guess this could be generalized to execute the build/install as any user.

**Comment by gcampbell-msft** [comment]
Closing in favor of tracking the various issues linked above.
